app/models/transaction.rb

```diff
@@ -1,3 +1,7 @@
 class Transaction < ApplicationRecord
+  validates_presence_of :credit_card_number
+  validates_presence_of :credit_card_expiration_date
```

Comment:
Even though the credit card attributes are strings, we can still validate for numericality because we don't expect any letters in a credit card number or expiration date: `numericality: true`